# Center For Systems Biology of Enteropathogens Sysbep

![Center For Systems Biology of Enteropathogens Sysbep](https://www.patricbrc.org/public/patric/images/SysBEP-1024x179.png)

[Systems Biology for Enteropathogens](http://www.sysbep.org/) uses a systems biology approach involving the use of iterative and complementary computational and experimental “omics” methodologies to analyze, identify, quantify, model, and ultimately predict the overall molecular processes involved in the pathogenesis of Salmonella and Yersinia species in macrophages.

The following are data sets generated by SysBEP with pointers to relevant publications about the data, pointers to the data, and sets of important genes/proteins identified by the set of experiments and the criteria for their selection.

## Salmonella

Sub-cellular proteomic survey of *Salmonella*

The project performed a subcellular proteomic analysis of *Salmonella enterica* subsp. *enterica serovar Typhimurium str. 14028* grown under standard laboratory and phagosome-mimicking conditions in vitro. Analysis of proteins from cytoplasmic, inner membrane, periplasmic, and outer membrane fractions yielded coverage of 25% of the theoretical proteome. Confident subcellular location could be assigned to over 1000 proteins, with good agreement between experimentally observed location and predicted/known protein properties. Comparison of protein location under the different environmental conditions provided insight into dynamic protein localization and possible moonlighting (multiple function) activities.

* Associated Publication(s):
  * A [Comprehensive Subcellular Proteomic Survey of Salmonella Grown under Phagosome-Mimicking versus Standard Laboratory Conditions](http://www.ncbi.nlm.nih.gov/pubmed/22900174)
* Protoemics Data [PeptideAtlas, accession PASS00551](http://www.peptideatlas.org/PASS/PASS00551)
* Genes of Interest:
  * [Cytoplasmic identified proteins](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_CYT.html)
  * [Inner membrane identified proteins](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_IM.html)
  * [Periplasmic identified proteins](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_PERI.html)
  * [Outer membrane identified proteins](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_OM.html)
  * [Promising moonlighting candidate proteins (known cytoplasmic proteins observed in the OM)](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_moonlighting.html)

Sample-matched multi-omic analysis of Salmonella regulatory mutants

Sample-matched multi-omic analysis of regulators required for virulence was extended to include metabolomics analysis. Here the transcriptome, proteome and metabolome of an additional six regulators required for virulence were analyzed in a sample-matched fashion to facilitate analysis of the Salmonella metabolic model.

* Transcriptomics Data Series GSE25441 at [GEO](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE25441) and at [PATRIC](https://www.patricbrc.org/view/ExperimentComparison/483361#view_tab=overview)

Systems analysis of *Salmonella* regulatory mutants

Sample-matched multi-omic measurements of fourteen virulence-essential regulator mutants were coupled with computational network analysis to efficiently identify *Salmonella* virulence factors.  A subset of computationally predicted virulence factors was determined to be secreted into the host cytoplasm, and two of these, SrfN and PagK2, were required for full mouse virulence and defined a new class of translocated effectors involved in pathogenesis.

* Associated Publication(s):
  * [Systems analysis of multiple regulator perturbations allows discovery of virulence factors in Salmonella](http://www.ncbi.nlm.nih.gov/pubmed/21711513)
* Proteomics Data [PNNL_1025](http://omics.pnl.gov/view/publication_1025.html)
* Transcriptomics Data Series GSE38378 at [GEO](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38378) and at [PATRIC](https://www.patricbrc.org/view/ExperimentComparison/867640#view_tab=overview)
* Genes of Interest:
  * [Broader list of pathogenesis related proteins](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_pathogenesis.html)
  * Network predicted virulence factors secreted into host cytoplasm
  * Validated translocated effectors required for virulence

*Salmonella* secretome analysis

Researchers identified effector proteins secreted into defined minimal medium designed to induce expression of the SPI-2 TTSS and its effectors. The secretome of the parent strain was compared to those of strains missing essential (ssaK::cat) or regulatory (ΔssaL) components of the SPI-2 TTSS.  20 known SPI-2 effectors were identified. To identify novel effector proteins, the secretome data was coupled with machine learning and 12 candidate proteins were selected for further characterization. Using CyaA’ reporter fusions, six novel type III effectors were identified and two additional proteins that were secreted into J774 macrophages independently of a TTSS were also identified.

* Associated Publication(s):
  * [Discovery of novel secreted virulence factors from Salmonella enterica serovar Typhimurium by proteomic analysis of culture supernatants](http://www.ncbi.nlm.nih.gov/pubmed/20974834)
* Proteomics Data: PNNL_1026
* Genes of Interest:
  * [Known SPI-2 effectors identified](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_SPI-2_effector.html)
  * [Novel secreted effectors identified](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_effector.html)

*Salmonella* growth condition adaptation

To determine the impact of a low Mg(2+)/pH defined growth medium (MgM) on the proteome of *Salmonella enterica* serotype Typhimurium, researchers cultured *S. Typhimurium* cells in the medium under two different conditions termed MgM Shock and MgM Dilution and then comparatively analyzed the bacterial cells harvested from these conditions by a global proteomic approach.

* Associated Publication(s):
  * [Proteome of Salmonella Enterica Serotype Typhimurium Grown in a Low Mg/pH Medium](http://www.ncbi.nlm.nih.gov/pubmed/19953200)
* Proteomics Data: [PNNL_0008](http://www.ncbi.nlm.nih.gov/pubmed/19953200)
* Genes of Interest:
  * [Proteins co-regulated with SPI-2 TTSS](https://www.patricbrc.org/webpage/website/data_collections/content/S_enterica_SPI-2_TTSS.html) (criteria – Pattern and magnitude of expression similar to SPI-2 TTSS across all conditions).

Experimental annotation of post-translational features and translated coding regions in the pathogen Salmonella Typhimurium

Complete and accurate genome annotation is crucial for comprehensive and systematic studies of biological systems. However, determining protein-coding genes for most new genomes is almost completely performed by inference using computational predictions with significant documented error rates (> 15%). Furthermore, gene prediction programs provide no information on biologically important post-translational processing events critical for protein function. We experimentally annotated the bacterial pathogen Salmonella Typhimurium 14028, using “shotgun” proteomics to accurately uncover the translational landscape and post-translational features.

* Associated Publication(s):
  * [Experimental annotation of post-translational features and translated coding regions in the pathogen Salmonella Typhimurium](http://www.ncbi.nlm.nih.gov/pubmed/21867535)
* Raw Data and Collated Peptide Identification Information: [PNNL_1039](http://www.ncbi.nlm.nih.gov/pubmed/21867535)

## Yersinia

Multi-omic analysis of Yersinia response to temperature shift

A temporal multi-omic analysis of *Y. pestis* and *Y. pseudotuberculosis* at physiologically relevant temperatures was performed to gain insights into how an acute and highly lethal bacterial pathogen, *Y. pestis*, differs from its less virulent progenitor, *Y. pseudotuberculosis*.

* Associated Publication(s):
  * [Comparative omics-driven genome annotation refinement: application across Yersiniae.](http://www.ncbi.nlm.nih.gov/pubmed/22479471)
  * [A multi-omic systems approach to elucidating Yersinia virulence mechanisms.](http://www.ncbi.nlm.nih.gov/pubmed/23147219)
* Proteomics Data [PNNL_1034](http://www.ncbi.nlm.nih.gov/pubmed/23147219)
* Transcriptomics Data Series GSE30634 at [GEO](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=gse30634) and at [PATRIC](https://www.patricbrc.org/view/ExperimentComparison/969586#view_tab=overview)
* Genes of Interest:
  * Up-regulated in Y. pestis CO92 relative to Y. pseudotuberculosis PB1/+ following a shift from flea vector to mammalian host temperature

*E. coli*

Evolved *E. coli* Data Consistent with Computed Optimal Growth

Differential gene and protein expression from wild-type and adaptively evolved strains supports observed growth phenotype changes, and is consistent with GEM-computed optimal growth states

* Associated Publication(s):
  * [Omic data from evolved E. coli are consistent with computed optimal growth from genome-scale models.](http://www.ncbi.nlm.nih.gov/pubmed/20664636)
* Proteomics Data [PNNL_1019](http://www.ncbi.nlm.nih.gov/pubmed/20664636)

## Thermotoga_maritima

*In silico* method for modelling metabolism and gene product expression at genome scale

A biochemically accurate model of molecular biology and metabolism will facilitate comprehensive and quantitative computations of an organism’s molecular constitution as a function of genetic and environmental parameters. A model of metabolism and macromolecular expression is prototyped using the simple microorganism *Thermotoga maritima*. The model accurately simulates variations in cellular composition and gene expression.

* Associated Publication(s):
  * [In silico method for modelling metabolism and gene product expression at genome scale.](http://www.ncbi.nlm.nih.gov/pubmed/22760628)
* Proteomics Data [PNNL_1063](http://www.ncbi.nlm.nih.gov/pubmed/22760628)

## Host Response Data

## Mouse Macrophage

Model-driven multi-omic data analysis elucidates metabolic immunomodulators of macrophage activation

Macrophages are central players in immune response, manifesting divergent phenotypes to control inflammation and innate immunity through release of cytokines and other signaling factors. Recently, the focus on metabolism has been reemphasized as critical signaling and regulatory pathways of human pathophysiology, ranging from cancer to aging, often converge on metabolic responses. Genome-scale modeling and multi-omics (transcriptomics, proteomics, and metabolomics) analysis are used to assess metabolic features that are critical for macrophage activation.

* Associated Publication(s):
  * [Model-driven multi-omic data analysis elucidates metabolic immunomodulators of macrophage activation](http://www.ncbi.nlm.nih.gov/pubmed/22735334)
* Metabolomics Data [PNNL_1066](http://www.ncbi.nlm.nih.gov/pubmed/22735334)
* Proteomics Data [PNNL_1065](http://www.ncbi.nlm.nih.gov/pubmed/22735334)
* Transcriptomics Data Series GSE35237 at [GEO](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE35237)

Proteomic investigation of the time course responses of RAW 264.7 macrophages to infection with *Salmonella enterica*

To investigate the extent to which macrophages respond to *Salmonella* infection, we infected RAW 264.7 macrophages with *Salmonella* enterica serotype Typhimurium and analyzed macrophage proteins at various time points following infection by using a global proteomic approach. A total of 1,006 macrophage and 115 *Salmonella* proteins were identified with high confidence.

* Associated Publication(s):
  * [Proteomic investigation of the time course responses of RAW 264.7 macrophages to infection with Salmonella enterica](http://www.ncbi.nlm.nih.gov/pubmed/19528222)
* Proteomics Data [PNNL_1009](http://omics.pnl.gov/view/publication_1009.html)

## Macaque

Characterization of macaque pulmonary fluid proteome during monkeypox infection: dynamics of host response

Understanding viral pathogenesis is challenging because of confounding factors, including nonabrasive access to infected tissues and high abundance of inflammatory mediators that may mask mechanistic details. In diseases such as influenza and smallpox where the primary cause of mortality results from complications in the lung, the characterization of lung fluid offers a unique opportunity to study host-pathogen interactions with minimal effect on infected animals. This investigation characterizes the global proteome response in the pulmonary fluid, bronchoalveolar lavage fluid, of macaques during upper respiratory infection by monkeypox virus (MPXV), a close relative of the causative agent of smallpox, variola virus.

* Associated Publication(s):
  * [Characterization of macaque pulmonary fluid proteome during monkeypox infection: dynamics of host response](http://www.ncbi.nlm.nih.gov/pubmed/20736407)
* Proteomics Data: [PNNL_1024](http://omics.pnl.gov/view/publication_1024.html)

