Acknowledgements
================

This page lists the data sources and third party software tools used in PATRIC. Click [here](./publications.html) for a list of PATRIC joint publications with other Scientific Collaborators.

## Data Sources Used by PATRIC

[ARDB (Antibiotic Resistance Genes Database)](http://ardb.cbcb.umd.edu/)

[BIND (Berkeley Internet Name Domain)](http://www.isc.org/software/bind/history)

[BioGRID (Biological General Repository for Interaction Datasets)](http://thebiogrid.org/)

[CARD (The Comprehensive Antibiotic Resistance
Database)](http://arpcard.mcmaster.ca/)

[CTD (Comparative Toxicogenomics Database)](http://ctdbase.org/)

[DIP (Database of Interacting
Proteins)](http://dip.doe-mbi.ucla.edu/dip/Main.cgi)

[DrugBank](http://v3.drugbank.ca/)

[EBI (European Bioinformatics Institute)](http://www.ebi.ac.uk/)

-   [ArrayExpress](http://www.ebi.ac.uk/arrayexpress/)
-   [PRIDE](http://www.ebi.ac.uk/pride/)
-   [IntAct](http://www.ebi.ac.uk/intact/)

[GAD (Genetic Association
Database)](http://geneticassociationdb.nih.gov/)

[GO (Gene Ontology)](http://www.geneontology.org/)

[HealthMap](http://www.healthmap.org/en/)

[HMP (Human Microbiome Project)](http://www.hmpdacc.org/)

[IEDB (Immune Epitope Database)](http://www.immuneepitope.org/)

[KEGG (Kyoto Encyclopedia of Genes and
Genomes)](http://www.genome.jp/kegg/)

[MINT (Molecular Interaction
Database)](http://mint.bio.uniroma2.it/mint/Welcome.do)

[NCBI (National Center for Biotechnology
Information)](http://www.ncbi.nlm.nih.gov/)

-   [BioProject](http://www.ncbi.nlm.nih.gov/bioproject/)
-   [CDD](http://www.ncbi.nlm.nih.gov/cdd/)
-   [Genome](http://www.ncbi.nlm.nih.gov/genome/)
-   [GEO](http://www.ncbi.nlm.nih.gov/geo/)
-   [MMDB](http://www.ncbi.nlm.nih.gov/Structure/MMDB/mmdb.shtml)
-   [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/)
-   [RefSeq](http://www.ncbi.nlm.nih.gov/RefSeq/)
-   [Taxonomy](http://www.ncbi.nlm.nih.gov/taxonomy)

[PDB (Protein Data Bank)](http://www.rcsb.org/pdb/home/home.do)

[PSICQUIC (Protemics Standard Initiative Common Query
Interface)](http://code.google.com/p/psicquic/)

[STITCH (Search Tool for Interactions of
Chemicals)](http://stitch.embl.de/cgi/show_input_page.pl?UserId=dew_Cq67PodO&sessionId=1xn5KWSzsTrZ)

[STRING (Search Tool for the Retrieval of Interacting
Genes/Proteins)](http://string.embl.de/newstring_cgi/show_input_page.pl?UserId=rVAoJJTi6kUC&sessionId=mt_5HRZP5HEI&info_box_type_input_page=general)

[TTD (Therapeutic Targets
Database)](http://bidd.nus.edu.sg/group/TTD/ttd.asp)

[UniProt (Universal Protein Resource)](http://www.uniprot.org/)

[Victors Virulence Factors](http://www.phidias.us/victors/index.php)

[VFDB (Virulence Factors Database)](http://www.mgc.ac.cn/VFs/)

## Software and Tools used by PATRIC

[BLAST (Basic Local Alignment Search
Tool)](http://blast.ncbi.nlm.nih.gov/Blast.cgi)

[BLAT](http://www.ncbi.nlm.nih.gov/pubmed/11932250)

[Circos](http://circos.ca/)

[Cluster3](http://bonsai.hgc.jp/~mdehoon/software/cluster/)

[Cytoscape](http://www.cytoscape.org/)

[FastTree](http://www.microbesonline.org/fasttree/)

[Gblocks](http://molevol.cmima.csic.es/castresana/Gblocks.html)

[HMMER](http://hmmer.janelia.org/)

[JBrowse](http://jbrowse.org/) via [The Holmes Lab](http://biowiki.org/)

[Jmol](http://jmol.sourceforge.net/)

[MCL (Markov Cluster Algorithm)](http://micans.org/mcl/)

[MUSCLE (Multiple Sequence Comparison by Log-
Expectation)](http://www.ebi.ac.uk/Tools/msa/muscle/)

[OrthoMCL](http://www.orthomcl.org/)


[PATRIC Data Release Policy](./data_release_policy.html)
