# Tutorial: Comprehensive Genome Analysis Service

## Overview
This set of short videos provides a step-by-step demonstration of using the PATRIC Comprehensive Genome Analysis service to assemble, annotate, and provide a basic comparative analysis of close relatives to a bacterial genome.

1. [Submitting a job using reads (13:08)](#cga1)
2. [Submitting contig files (3:36)](#cga2)
3. [Full genome report (13:06)](#cga3)
4. [Job results (11:22)](#cga4)
5. [Assembly results (6:13)](#cga5)
6. [Annotation results (8:50)](#cga6)
7. [Viewing your genome (8:05)](#cga7)
8. [Finding help and citing PATRIC (6:51)](#cga8)


### See also
* [Comprehensive Genome Analysis Service](https://patricbrc.org/app/ComprehensiveGenomeAnalysis)
* [Comprehensive Genome Analysis Service Tutorial](https://docs.patricbrc.org/tutorial/comprehensive-genome-analysis/comprehensive-genome-analysis2.html) (document)

## Using the Comprehensive Genome Analysis Service

Links to the videos are provided below.  They can also be viewed in order in the [PATRIC YouTube channel playlist](https://www.youtube.com/playlist?list=PLsstVALeacEL_UE2oYbJW45BDHmwL9cjz). Relevant references are provided at the bottom of this page.

<a name="cga1"></a>
##### 1. Submitting a job using reads (13:08)
<iframe width="560" height="315" src="https://www.youtube.com/embed/8hU2t4OhMCA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga2"></a>
##### 2. Submitting contig files (3:36)
<iframe width="560" height="315" src="https://www.youtube.com/embed/NquDD9LoTdc" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga3"></a>
##### 3. Full genome report (13:06)
<iframe width="560" height="315" src="https://www.youtube.com/embed/XqgyobAw2sQ" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga4"></a>
##### 4. Job results (11:22)
<iframe width="560" height="315" src="https://www.youtube.com/embed/_NgplRiYfJw" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga5"></a>
##### 5. Assembly results (6:13)
<iframe width="560" height="315" src="https://www.youtube.com/embed/_aqN_V12VZk" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga6"></a>
##### 6. Annotation results (8:50)
<iframe width="560" height="315" src="https://www.youtube.com/embed/bqkNsg_QqIM" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga7"></a>
##### 7. Viewing your genome (8:05)
<iframe width="560" height="315" src="https://www.youtube.com/embed/_C-a-UG92Ws" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="cga8"></a>
##### 8. Finding help and citing PATRIC (6:51)
<iframe width="560" height="315" src="https://www.youtube.com/embed/k65b1fCsSEk" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## References
* Antipov, D., et al., plasmidSPAdes: assembling plasmids from whole genome sequencing data. bioRxiv, 2016: p. 048942.
* Bankevich, A., et al., SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing. Journal of computational biology, 2012. 19(5): p. 455-477.
* Brettin, T., et al., RASTtk: a modular and extensible implementation of the RAST algorithm for building custom annotation pipelines and annotating batches of genomes. Scientific reports, 2015. 5: p. 8365.
* Davis, J.J., et al., PATtyFams: Protein families for the microbial genomes in the PATRIC database. 2016. 7: p. 118.
* Edgar, R.C., MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic acids research, 2004. 32(5): p. 1792-1797.
* Koren, S., et al., Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation. Genome research, 2017. 27(5): p. 722-736.
* Krueger, F., Trim Galore: a wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries. URL http://www. bioinformatics. babraham. ac. uk/projects/trim_galore/.(Date of access: 28/04/2016), 2012.
* Nurk, S., et al., metaSPAdes: a new versatile metagenomic assembler. Genome research, 2017. 27(5): p. 824-834.
* Ondov, B.D., et al., Mash: fast genome and metagenome distance estimation using MinHash. Genome biology, 2016. 17(1): p. 132.
* Overbeek, R., et al., The SEED and the Rapid Annotation of microbial genomes using Subsystems Technology (RAST). Nucleic Acids Res, 2014. 42(Database issue): p. D206-14.
* Stamatakis, A., RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 2014. 30(9): p. 1312-1313.
* Vaser, R., et al., Fast and accurate de novo genome assembly from long uncorrected reads. Genome research, 2017. 27(5): p. 737-746.
* Walker, B.J., et al., Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement. PloS one, 2014. 9(11): p. e112963.
* Wick, R.R., et al., Unicycler: resolving bacterial genome assemblies from short and long sequencing reads. PLoS computational biology, 2017. 13(6): p. e1005595.
