# Tutorial: Genome Assembly Service

## Overview
This set of short videos provides a step-by-step demonstration of using the PATRIC Genome Assembly Service to create an assembled genome from a set a bacterial sequence reads.

1. [Uploading read files to the workspace (11:08)](#assembly1)
2. [Submitting reads to the Genome Assembly Service (9:15)](#assembly2)
3. [Selecting assembly strategies and submitting the assembly job (7:14)](#assembly3)
4. [Monitoring the assembly job and reviewing job results (8:06)](#assembly4)
5. [Assembly questions and getting help with problems (4:16)](#assembly5)

### See also
* [Genome Assembly Service](https://patricbrc.org/app/Assembly2)
* [Genome Assembly Service Tutorial](https://docs.patricbrc.org/tutorial/genome_assembly/assembly2.html) (document)

## Using the Genome Assembly Service

Links to the videos are provided below.  They can also be viewed in order in the [PATRIC YouTube channel playlist](https://www.youtube.com/playlist?list=PLsstVALeacEIk2iu794_XYrPFUEAETy_l). Relevant references are provided at the bottom of this page.

<a name="assembly1"></a>
### 1. Uploading read files to the workspace (11:08)
<iframe width="560" height="315" src="https://www.youtube.com/embed/aCJrMDyfDDk" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="assembly2"></a>
### 2. Submitting reads to the Genome Assembly Service (9:15) 
<iframe width="560" height="315" src="https://www.youtube.com/embed/89cpfkQvgWI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="assembly3"></a>
### 3. Selecting assembly strategies and submitting the assembly job (7:14)
<iframe width="560" height="315" src="https://www.youtube.com/embed/afxnUM7jtk4" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="assembly4"></a>
### 4. Monitoring the assembly job and reviewing job results (8:06)
<iframe width="560" height="315" src="https://www.youtube.com/embed/wWq1Ec6yhWM" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="assembly5"></a>
### 5. Assembly questions and getting help with problems (4:16)
<iframe width="560" height="315" src="https://www.youtube.com/embed/e8dnNw0Ir0w" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## References
* Antipov, D., et al., plasmidSPAdes: assembling plasmids from whole genome sequencing data. bioRxiv, 2016: p. 048942.
* Bankevich, A., et al., SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing. Journal of computational biology, 2012. 19(5): p. 455-477.
* Gurevich, A., et al., QUAST: quality assessment tool for genome assemblies. Bioinformatics, 2013. 29(8): p. 1072-1075.
* Koren, S., et al., Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation. Genome research, 2017. 27(5): p. 722-736.
* Krueger, F., Trim Galore: a wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries. URL http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/ (Date of access: 28/04/2016), 2012.
* Nurk, S., et al., metaSPAdes: a new versatile metagenomic assembler. Genome research, 2017. 27(5): p. 824-834.
* Vaser, R., et al., Fast and accurate de novo genome assembly from long uncorrected reads. Genome research, 2017. 27(5): p. 737-746.
* Walker, B.J., et al., Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement. PloS one, 2014. 9(11): p. e112963.
* Wick, R.R., et al., Bandage: interactive visualization of de novo genome assemblies. Bioinformatics, 2015. 31(20): p. 3350-3352.
* Wick, R.R., et al., Unicycler: resolving bacterial genome assemblies from short and long sequencing reads. PLoS computational biology, 2017. 13(6): p. e1005595.

