# Tutorial: Proteome Comparison Service

## Overview
This set of short videos provides a step-by-step demonstration of using the PATRIC Proteome Comparison Service to identify insertions and deletions in up to nine target genomes that are compared with one reference, which can be a researcherâ€™s private genome in PATRIC, a genome that has been annotated outside PATRIC, any of the publicly available genomes in PATRIC, or a set of proteins that you have saved in PATRIC as a feature group.

1. [Selecting parameters (8:13)](#proteome1)
2. [Selecting a reference genome (3:05)](#proteome2)
3. [Using a feature group or FASTA file as input (5:20)](#proteome3)
4. [Using a FASTA file as the reference (3:38)](#proteome4)
5. [Using a feature group as the reference (2:04)](#proteome5)
6. [Selecting the genomes to use for the comparison (5:57)](#proteome6)
7. [3 flavors of reference genomes (5:38)](#proteome7)
8. [Reviewing the output job results (22:54)](#proteome8)
9. [Citing PATRIC and getting help (5:52)](#proteome9)

### See also
* [Proteome Comparison Service](https://patricbrc.org/app/SeqComparison)
* [Proteome Comparison Service Tutorial](https://docs.patricbrc.org/tutorial/proteome_comparison/proteome_comparison.html) (document)

## Using the Proteome Comparison Service

Links to the videos are provided below.  They can also be viewed in order in the [PATRIC YouTube channel playlist](https://www.youtube.com/watch?v=hV_cnv0icb4&list=PLsstVALeacEIh8Tq9tqAJcWpXySlAcTP_). Relevant references are provided at the bottom of this page.

<a name="proteome1"></a>
##### 1. Selecting parameters (8:13)
<iframe width="560" height="315" src="https://www.youtube.com/embed/hV_cnv0icb4" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome2"></a>
##### 2. Selecting a reference genome (3:05) 
<iframe width="560" height="315" src="https://www.youtube.com/embed/IpU84kXqKkI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome3"></a>
##### 3. Using a feature group or FASTA file as input (5:20)
<iframe width="560" height="315" src="https://www.youtube.com/embed/8PyZCkGqku8" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome4"></a>
##### 4. Using a FASTA file as the reference (3:38)
<iframe width="560" height="315" src="https://www.youtube.com/embed/PtdJ0YDkssA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome5"></a>
##### 5. Using a feature group as the reference (2:04)
<iframe width="560" height="315" src="https://www.youtube.com/embed/Hen1vQr6s3Q" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome6"></a>
##### 6. Selecting the genomes to use for the comparison (5:57)
<iframe width="560" height="315" src="https://www.youtube.com/embed/yzT6awJDiok" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome7"></a>
##### 7. 3 flavors of reference genomes (5:38)
<iframe width="560" height="315" src="https://www.youtube.com/embed/_7w1foW5whU" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome8"></a>
##### 8. Reviewing the output job results (22:54)
<iframe width="560" height="315" src="https://www.youtube.com/embed/dND2JAAdsQI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

<a name="proteome9"></a>
##### 9. Citing PATRIC and getting help (5:52)
<iframe width="560" height="315" src="https://www.youtube.com/embed/bWvGddaYYlg" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## References
* Overbeek, R., et al., The SEED and the Rapid Annotation of microbial genomes using Subsystems Technology (RAST). Nucleic acids research, 2014. 42(D1): p. D206-D214.

